compiler/src/dmd/targetcompiler.d

```diff
@@ -0,0 +1,210 @@
+/**
+ * Encapsulates the vagaries of which of the three D compilers (gdc, ldc or dmd) is being built
+ *
+ * Copyright:   Copyright (C) 1999-2025 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 https://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 https://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/compiler/src/dmd/mars.d, _targetcompiler.d)
+ * Documentation:  https://dlang.org/phobos/dmd_targetcompiler.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/compiler/src/dmd/targetcompiler.d
+ */
+
+module dmd.targetcompiler;
+
+version (IN_GCC) {}        // compiler is being built with gdc
+else version (IN_LLVM) {}  // compiler is being built with ldc
+else version = MARS;       // default means compiler is built with Digital Mars compiler (DMD)
+
+/***************************
+ */
+
+version (IN_GCC)
+{
+    enum TargetCompiler = "GNU D";    // compiler being built
+    enum SwitchPrefix = "f";          // prefix to switch
+    enum SwitchVariadic = "fno-rtti"; // disables D-style variadic functions
+    enum const(char)* SwitchExceptions = "fno-exceptions"; // switch that disables exceptions
+    enum const(char)* SwitchScopeGuard = "fno-exceptions"; // switch that disables exceptions
```

Comment:
I wonder whether an enum array of option strings and and enum for indexing into the array by name would be better as a follow-up PR.

Or perhaps moving all bool params such as `useTypeInfo` to a mask value, hitting two with one stone.

A new overload could even be added accepting said enum values for errors.
```
error(OPTnothrow, tcs.loc, "cannot use try-catch statements");
```
Which would append the corresponding CLI option that enabled the diagnostic at the end of the line internally. Rather than have the caller name CLI options that don't exist in gdc and ldc.

Reply:
In the meantime, I think the names aren't too clear, and propos to rename `SwitchVariadic` to `SwitchNoTypeInfos`, `SwitchExceptions` to `SwitchNoExceptions`. `SwitchScopeGuard` isn't used at all.

Reply:
I understand that and agree, but here we aren't talking about bools, but strings, and those strings happen to contain `no-`, because the user is opting out of default TypeInfos, exceptions, ModuleInfos…